File: tests/fixtures/go-app/calc.go
// Package calc is the toy library exercised by calc_test.go.
// It exists purely so reusable-ci.yml has a Go code path to dog-food.
package calc

// Add returns the sum of a and b.
func Add(a, b int) int {
	return a + b
}

// Greet returns a simple greeting for name.
func Greet(name string) string {
	return "hello, " + name
}

File: tests/fixtures/java-app/src/main/java/io/github/citemplates/selftest/Calc.java
package io.github.citemplates.selftest;

/**
 * Trivial helpers exercised by {@code CalcTest}. Exist solely so
 * reusable-ci.yml has a Java code path to dog-food.
 */
public final class Calc {

    private Calc() {
        // utility class, no instantiation
    }

    public static int add(int a, int b) {
        return a + b;
    }

    public static String greet(String name) {
        return "hello, " + name;
    }
}
